app/models/work.rb

```diff
@@ -0,0 +1,36 @@
+class Work < ApplicationRecord
+  # relation - user can have many votes
+  has_many :votes, dependent: :destroy
+  # validate for unique title
+  validates :title, presence: true, uniqueness: true
+
+  def self.order_works_by_votes
+    works = Work.all
+    ordered_works = works.sort_by { |work| work.votes.size }.reverse
+    return ordered_works
+  end
+
+  def self.top_vote_getter
+    return Work.order_works_by_votes.first
+  end
+
+  def self.top_vote_getter_votes
+    return Work.top_vote_getter.votes.size
+  end
```

Comment:
I love the way the methods in this file build on each other, each providing a small bit of functionality. This is an excellent example of functional decomposition.